Release

jonasscheid opened the pull request:
A lot of template merges are hiding in dev branch since 2022: https://github.com/nf-core/hlatyping/commit/104998212bc49caa394005020e2d752c525445ec 

Lets do a release @christopher-mohr @apeltzer.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/hlatyping/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/hlatyping _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

nf-core-bot committed:
Template update for nf-core/tools version 2.7.1

nf-core-bot committed:
Template update for nf-core/tools version 2.7.2

christopher-mohr committed:
Merge branch 'TEMPLATE' of https://github.com/nf-core/hlatyping into merge_template_2_7_2

christopher-mohr committed:
update gunzip and yara module

christopher-mohr committed:
bump nextflow version for ci

christopher-mohr committed:
Merge pull request #152 from christopher-mohr/merge_template_2_7_2

Merge template updates

nf-core-bot committed:
Template update for nf-core/tools version 2.8

nf-core-bot committed:
Template update for nf-core/tools version 2.9

nf-core-bot committed:
Template update for nf-core/tools version 2.10

nf-core-bot committed:
Template update for nf-core/tools version 2.11

nf-core-bot committed:
Template update for nf-core/tools version 2.11.1

nf-core-bot committed:
Template update for nf-core/tools version 2.12

nf-core-bot committed:
Template update for nf-core/tools version 2.13

nf-core-bot committed:
Template update for nf-core/tools version 2.13.1

christopher-mohr committed:
update optitype module

christopher-mohr committed:
add both output channels

christopher-mohr committed:
Merge branch 'TEMPLATE' of https://github.com/nf-core/hlatyping into template_merge_2_13_1

christopher-mohr committed:
fix branching

christopher-mohr committed:
fix ci.yml

christopher-mohr committed:
remove trailing whitespace

christopher-mohr committed:
change default value

christopher-mohr committed:
fix linting warnings

christopher-mohr committed:
update CHANGELOG

christopher-mohr committed:
changes based on review

christopher-mohr committed:
Update nextflow_schema.json

Co-authored-by: Friederike Hanssen <Friederike.hanssen@qbic.uni-tuebingen.de>

christopher-mohr committed:
remove fasta param from nextflow.config

christopher-mohr committed:
Merge pull request #165 from christopher-mohr/template_merge_2_13_1

Template merge

christopher-mohr committed:
Merge branch 'dev' into update_optitype_module

christopher-mohr committed:
fix syntax error

christopher-mohr committed:
Merge pull request #164 from christopher-mohr/update_optitype_module

Update optitype module

nf-core-bot committed:
Template update for nf-core/tools version 2.14.0

nf-core-bot committed:
Template update for nf-core/tools version 2.14.1

SpikyClip committed:
Add missing --enumerate arg.

SpikyClip committed:
Add dedicated optitype output folder.

christopher-mohr committed:
Merge branch 'TEMPLATE' of https://github.com/nf-core/hlatyping into merge-template-2-14-1

christopher-mohr committed:
Merge pull request #171 from christopher-mohr/merge-template-2-14-1

Merge latest template update

SpikyClip committed:
Merge branch 'dev' into optitype_enumerate_fix

SpikyClip committed:
Merge branch 'dev' into optitype_dir

christopher-mohr committed:
Merge pull request #170 from SpikyClip/optitype_dir

Add dedicated optitype output folder.

christopher-mohr committed:
Merge pull request #169 from SpikyClip/optitype_enumerate_fix

Add missing --enumerate arg.

nf-core-bot committed:
Template update for nf-core/tools version 3.0.0

nf-core-bot committed:
Template update for nf-core/tools version 3.0.1

nf-core-bot committed:
Template update for nf-core/tools version 3.0.2

nf-core-bot committed:
Template update for nf-core/tools version 3.1.0

nf-core-bot committed:
Template update for nf-core/tools version 3.1.1

nf-core-bot committed:
Template update for nf-core/tools version 3.1.2

SpikyClip committed:
Installs cat fastq module using nf-core package.

SpikyClip committed:
Installs nf-schema@2.3.0 plugin.

Has more useful parsing/validation methods, which would be relevant as this pipeline takes multiple possible data types (i.e. bams and/or fastqs).

SpikyClip committed:
Adds subworkflow to test inputs after runs of the same sample are concatenated in channel.

Logic borrowed from nf-core rnaseq:
https://github.com/nf-core/rnaseq/blob/b96a75361a4f1d49aa969a2b1c68e3e607de06e8/subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf#L158-L174

SpikyClip committed:
Adds cat_fastq logic to main workflow.

meta.id is appended to front of list to facilitate joining by `.groupTuple()`, and is eventually dropped by `checkSamplesAfterGrouping` which returns `[meta, input_files]` as expected by rest of pipeline.

SpikyClip committed:
Migrates to json schema 2020-12, removing native groovy tests and replacing with tests in schema_input.json.

SpikyClip committed:
Adds parameter to save merged fastqs.

SpikyClip committed:
Adds test profile for concatenation of multiple fastq files.

SpikyClip committed:
Adds test_fastq_cat as a test profile.

SpikyClip committed:
Updates usage, README, and output to include `cat_fastq` feature.

SpikyClip committed:
Merge branch 'dev' into cat_fastq_feature

Apologies, should have worked off `dev` to begin with.

nf-core-bot committed:
Template update for nf-core/tools version 3.2.0

christopher-mohr committed:
Merge branch 'TEMPLATE' of https://github.com/nf-core/hlatyping into merging-template-updates-3-2

christopher-mohr committed:
add references

christopher-mohr committed:
remove duplicate subworkflow

christopher-mohr committed:
remove duplicate schema key

christopher-mohr committed:
increase gitpod cpus

christopher-mohr committed:
add resourceLimits in test configs

christopher-mohr committed:
update CHANGELOG

christopher-mohr committed:
add contributors information

christopher-mohr committed:
Merge pull request #181 from christopher-mohr/merging-template-updates-3-2

Merging template updates up to 3.2

jonasscheid committed:
Merge branch 'dev' into cat_fastq_feature

jonasscheid committed:
Update nextflow.config

SpikyClip committed:
Update README.md

Simplify language

Co-authored-by: Christopher Mohr <contact.cmohr@gmail.com>

SpikyClip committed:
Update conf/test_fastq_cat.config

Use params.pipelines_testdata_base_path for new test samplesheet path.

Co-authored-by: Christopher Mohr <contact.cmohr@gmail.com>

SpikyClip committed:
Update conf/test_fastq_cat.config

Complete config_profile_name

Co-authored-by: Christopher Mohr <contact.cmohr@gmail.com>

SpikyClip committed:
Update docs/output.md

Simplify language

Co-authored-by: Christopher Mohr <contact.cmohr@gmail.com>

SpikyClip committed:
Fixes bug where paired end files were detected as fastqs to concatenate.

SpikyClip committed:
Removes unused module.

Otherwise nf-core pipeline lint keeps reinstalling it.

SpikyClip committed:
Fixes linting error where optional_outputs was missing from allOf key.

mapo9 committed:
fix linting errors

mapo9 committed:
remove duplicate lines

mapo9 committed:
multiqc config update

mapo9 committed:
multiqc config update. this time correct (?)

mapo9 committed:
Merge pull request #188 from mapo9/new_template

New template

mapo9 committed:
module bumb

mapo9 committed:
fixed problems with new modules

mapo9 committed:
collatefastq reinstall to update to most recent

mapo9 committed:
better comment on interleaved

mapo9 committed:
Merge pull request #189 from mapo9/module_bump

Module bump

mapo9 commented:
@christopher-mohr , @jonasscheid would be great if you could add your review

maxulysse commented on CITATIONS.md:
```diff
@@ -12,6 +12,8 @@
 
 - [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
 
+> Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].
```
```suggestion
  > Andrews, S. (2010). FastQC: A Quality Control Tool for High Throughput Sequence Data [Online].
```

maxulysse commented on CITATIONS.md:
```diff
@@ -27,7 +29,8 @@
 - [MultiQC](https://pubmed.ncbi.nlm.nih.gov/27312411/)
-  > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
+
+> Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```
```suggestion
  > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```

maxulysse replied:
That was not fixed.

maxulysse commented on README.md:
```diff
@@ -70,6 +89,8 @@ For further information or help, don't hesitate to get in touch on the [Slack `#
 
 If you use nf-core/hlatyping for your analysis, please cite it using the following doi: [10.5281/zenodo.1401039](https://doi.org/10.5281/zenodo.1401039)
 
+<!-- TODO nf-core: Add bibliography of tools and data used in your pipeline -->
+
```
```suggestion
```

maxulysse commented on README.md:
```diff
@@ -21,40 +28,52 @@ On release, automated continuous integration tests run the pipeline on a full-si
 
 ## Pipeline summary
 
+<!-- TODO nf-core: Include a figure that guides the user through the major workflow steps. Many nf-core
+     workflows use the "tube map" design for that. See https://nf-co.re/docs/contributing/design_guidelines#examples for examples.   -->
+
```
```suggestion
```

maxulysse commented on assets/methods_description_template.yml:
```diff
@@ -3,16 +3,21 @@ description: "Suggested text and references to use when describing pipeline usag
 section_name: "nf-core/hlatyping Methods Description"
 section_href: "https://github.com/nf-core/hlatyping"
 plot_type: "html"
+## TODO nf-core: Update the HTML below to your preferred methods description, e.g. add publication citation for this pipeline
```
```suggestion
```

maxulysse commented on main.nf:
```diff
@@ -4,58 +4,96 @@
-WorkflowMain.initialise(workflow, params, log)
+//   This is an example of how to use getGenomeAttribute() to fetch parameters
+//   from igenomes.config using `--genome`
+//params.fasta = getGenomeAttribute('fasta')
```
how do you deal with the references?

mapo9 replied:
This was commented out already when I started working on the pipeline. I checked and didnt find a module that was using a reference, so I assume we dont need it. However, the plan is to add more tools once the release is done. I kept it commented out so I dont forget how to call it when other modules will need it. But I think we can also remove for now

christopher-mohr replied:
The pipeline (currently) uses specific HLA references only (https://github.com/nf-core/hlatyping/tree/master/data/references). We can delete this for now.

maxulysse commented on tower.yml:
Try to populate this with the small files created by the pipeline

maxulysse reviewed (changes requested):
Looking good, minor comments

christopher-mohr commented:
> @christopher-mohr , @jonasscheid would be great if you could add your review

I will.

christopher-mohr commented on main.nf:
```diff
@@ -4,58 +4,96 @@
-WorkflowMain.initialise(workflow, params, log)
+//   This is an example of how to use getGenomeAttribute() to fetch parameters
+//   from igenomes.config using `--genome`
+//params.fasta = getGenomeAttribute('fasta')
```
```suggestion
```

christopher-mohr commented on CHANGELOG.md:
```diff
@@ -3,6 +3,22 @@
+## 2.1.0dev - [] - 2022-12-22
+
+### `Added`
+
```
```suggestion

- [#169](https://github.com/nf-core/hlatyping/pull/169) - Parameter `enumerations` to specify the number of solutions provided by Optitype
```

christopher-mohr reviewed (commented):
LGTM overall. Might make sense to wait for the [lane-merging feature](https://github.com/nf-core/hlatyping/pull/179) for this release.

SpikyClip committed:
Merge branch 'dev' into cat_fastq_feature

SpikyClip committed:
Fixes bug caused by unclosed comment.

SpikyClip committed:
Fixes bug caused by wrong channel name.

For clarity, move channel mixing to one point in script.

SpikyClip committed:
Adds missing ending newline.

SpikyClip committed:
Eliminates bug where pipeline fails if only one sample is in samplesheet.

Caused by ifEmpty returning null which propagates down the version channels causing issues.

zachary-foster commented on README.md:
```diff
@@ -1,18 +1,25 @@
 
-**nf-core/hlatyping** is a bioinformatics best-practice analysis pipeline for Precision HLA typing from next-generation sequencing data.
+**nf-core/hlatyping** is a bioinformatics pipeline that can be used to perform HLA typing from next-generation sequencing data.
 The pipeline does next-generation sequencing-based Human Leukozyte Antigen (HLA) typing using [OptiType](https://github.com/FRED-2/OptiType). OptiType is a HLA genotyping algorithm based on integer linear programming. Reads of whole exome/genome/transcriptome sequencing data are mapped against a reference of known MHC class I alleles. To produce accurate 4-digit HLA genotyping predictions, all ma ...
```
```suggestion
The pipeline does next-generation sequencing-based Human Leukocyte Antigen (HLA) typing using [OptiType](https://github.com/FRED-2/OptiType). OptiType is a HLA genotyping algorithm based on integer linear programming. Reads of whole exome/genome/transcriptome sequencing data are mapped against a reference of known MHC class I alleles. To produce accurate 4-digit HLA genotyping predictions, all major and minor HLA-I loci are considered simultaneously to find an allele combination that maximizes the number of explained reads.
```

zachary-foster commented on conf/modules.config:
```diff
@@ -52,11 +36,21 @@ process {
 
     withName: OPTITYPE {
         publishDir = [
-                path: { "${params.outdir}" },
+                path: { "${params.outdir}/optitype" },
                 mode: params.publish_dir_mode,
                 saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
             ]
```
This looks the same as the default on line 15. Probably can delete with no effect.

zachary-foster commented on conf/modules.config:
```diff
@@ -52,11 +36,21 @@ process {
 
+    withName: 'MULTIQC' {
+        ext.args   = { params.multiqc_title ? "--title \"$params.multiqc_title\"" : '' }
+        publishDir = [
+            path: { "${params.outdir}/multiqc" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
```
Same here

zachary-foster reviewed (approved):
Looks good to me! I left a few minor comments

mapo9 committed:
dealing with minor review comments

mapo9 committed:
update changelog

mapo9 committed:
populate tower yml

mapo9 committed:
prettier

mapo9 committed:
Merge pull request #190 from mapo9/review_changes

Review changes

SpikyClip committed:
Merge branch 'dev' into cat_fastq_feature

jonasscheid committed:
Update test_fastq_cat.config

jonasscheid committed:
Merge pull request #179 from SpikyClip/cat_fastq_feature

Cat fastq feature

jonasscheid committed:
Update CHANGELOG.md

jonasscheid committed:
Update contributors

jonasscheid committed:
fix typo

mapo9 committed:
Merge pull request #191 from nf-core/update-changelog

Update changelog and contributors

jonasscheid commented:
@maxulysse requested changes were implemented by @mapo9, can you take another look? Thanks!

mapo9 committed:
fixed citation

maxulysse committed:
Merge pull request #192 from mapo9/fix_multiqc_cite

fixed multiqc citation
